### ISSUE TITLE
downsampling for tracking doesn't work properly

### ISSUE THREAD

**Issue by mixmixmix** [ctorney/uavTracker#17]

**Comment by aakanksharathore** [comment]
step_frames was set to 15 earlier but somehow it was still working fine. I have pushed the latest changes to the bb_tracking branch.  I don't understand how it was giving me the output for full 5 minutes at 30 FPS. If it was 15 steps then we might get only 20 seconds long output, no?
I think something is wrong, maybe the stop condition was not working? I have nested the if statement and pushed it to my branch. can you test this latest code on any one video? We will just see if it is working fine, it will take too long time in my system but I have started a test video anyway.

**Comment by mixmixmix** [comment]
I'm testing in my branch https://github.com/ctorney/uavTracker/tree/me-proof-framework, where I improve the overall framework. There is something clearly wrong. I'm testing on some short videos, and once I found the issue I will commit fix to your branch separately.

**Comment by aakanksharathore** [comment]
It wasn't working because of this "and" condition in first if statement-

                if ((i%step_frames)!=0) and (i < period["start"]):
                    continue
                if (i > period["stop"]) and (period["stop"] != 0):

It should be OR, right?

**Comment by mixmixmix** [comment]
Yes, it should be `or` :} My mistake, sorry! 

~~I tested it a bit and I think it only means that all the runs you did so far were tracking all the frames instead of skipping any, so data is correct.~~

There would be some tracking of the period before specified ?

**Comment by mixmixmix** [comment]
If you want to change it in your branch, could you also remove `frame_idx` and replace it with `i`? In the tracks so far the `frame_idx` would refer to the beginning of the period, not the file (which I think is wrong)

**Comment by aakanksharathore** [comment]
No issues :-D
Yes, I am glad that I dont have to rerun those videos. I will make this
correction in my branch.

**Comment by aakanksharathore** [comment]
Yeah, I have already done that. I am assigning i to frame_idx

**Comment by mixmixmix** [comment]
But you are assigning it after it is used in the iteration, I think it might get a wrong value. Better to get rid of it completely becasue it should be the same as `i`

**Comment by mixmixmix** [comment]
Can you check your output video files? What I expect is that they are at first sped up, and then at normal speed, but the tracking values might be not corresponding to the correct frames. (Though it should be possible to work out when they are wrong)

I have some problems with transformations in my test file, so I will work a bit on a tracker before re-running on server.

**Comment by mixmixmix** [comment]
The condition should actually look like following

```
                #jump frames
                if i > period["stop"] and period["stop"] != 0:
                    break
                if i < period["start"]:
                    continue
                if ((i - period["start"]) % step_frames):
                    continue
```

I will keep testing on my test rockinghorse setup. I should have tested it thoroughly before we started running your videos - sorry about that.

**Comment by mixmixmix** [comment]
I should have fixed all issues in this commit: https://github.com/ctorney/uavTracker/commit/1f02a00fad412be7ceda3d03d51122fbe3ff653c

**Comment by mixmixmix** [comment]
@aakanksharathore, I will finish my branch and test it, and will make sure all the changes in yours are also taken into account. Because I did a lot of style edits etc. it might not be possible to merge my changes so it might make sense if you then create a new branch _After_ I push my changes.

**Comment by aakanksharathore** [comment]
Ok, let me know once the changes are done, I will create branch after that.


*Namastey!*

Akanksha Rathore
PhD scholar
CES, Indian Institute of Science
Bangalore 560012
website: https://sites.google.com/view/akanksha-rathore
twitter: akanksha_ak5 <https://twitter.com/akanksha_ak5>

-------------------------------------------------------------------------------------------------------------
सर्वे भवन्तु सुखिनः
सर्वे सन्तु निरामयाः ।
सर्वे भद्राणि पश्यन्तु
मा कश्चिद्दुःखभाग्भवेत् ।
ॐ शान्तिः शान्तिः शान्तिः ॥

**Comment by mixmixmix** [comment]
> Can you check your output video files? What I expect is that they are at first sped up, and then at normal speed, but the tracking values might be not corresponding to the correct frames. (Though it should be possible to work out when they are wrong)

To clarify: because the condition was `and`, there was tracking of every ’step_size’ frame for all the frames before `period[start]`. So if `step_size=15` then we’d expect to have an output video with 2fps initially and later when it tracks the frames in the period, to be 30fps. That should also show up by having more frames in output txt tracks file, so if you could send me them I can try to work it out.

**Comment by aakanksharathore** [comment]
You are right, output videos are indeed fast in the beginning and then
normal speed for the specified duration. The problem is that till now we
were not writing actual frame number in the outpyt file so we cant see this
pattern in text files. Will send you output files in the morning they are
on another system in the lab.
